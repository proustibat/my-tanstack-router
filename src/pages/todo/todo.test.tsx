import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TodoApp from "./index";

describe("TodoApp", () => {
    it("renders the init tasks", () => {
        // since screen does not have the container property, we'll destructure render to obtain a container for this test
        const { container } = render(<TodoApp />);
        expect(container).toMatchSnapshot();
    });

    it("renders correct heading", () => {
        render(<TodoApp />);
        expect(screen.getByRole("heading", { level: 1 }).textContent).toMatch(
            /TODO/i,
        );
    });

    it("deletes a task", async () => {
        const user = userEvent.setup();
        render(<TodoApp />);
        expect(screen.queryByText(/Youpi/i)).toBeInTheDocument();
        const allDeleteButtons = screen.getAllByText(/delete/i);
        expect(allDeleteButtons).toHaveLength(4);
        await user.click(allDeleteButtons[0]);
        expect(screen.queryByText(/Youpi/i)).not.toBeInTheDocument();
        expect(screen.getAllByText(/delete/i)).toHaveLength(3);
    });
});