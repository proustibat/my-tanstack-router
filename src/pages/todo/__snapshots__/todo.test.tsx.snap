// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoApp > renders the init tasks 1`] = `
<div>
  <div>
    <h1
      class="_title_a1a131"
    >
      TODO
    </h1>
    <main
      class="_main_a1a131"
    >
      <section>
        <h3>
          App with class components
        </h3>
        bound HTMLFormElement {
          "0": <input
            name="task-entry"
            type="text"
            value=""
          />,
          "1": <button
            disabled=""
            type="submit"
          >
            Submit
          </button>,
        }
        <h4>
          All the tasks!
          <span>
             (
            2
            ) 
          </span>
        </h4>
        <ul>
          <li>
            <span>
              Youpi
            </span>
            <button
              type="button"
            >
              Edit
            </button>
            <button
              type="button"
            >
              delete
            </button>
          </li>
          <li>
            <span>
              Yep
            </span>
            <button
              type="button"
            >
              Edit
            </button>
            <button
              type="button"
            >
              delete
            </button>
          </li>
        </ul>
      </section>
      <section>
        <h3>
          App with functional components
        </h3>
        bound HTMLFormElement {
          "0": <input
            name="task-entry"
            type="text"
            value=""
          />,
          "1": <button
            disabled=""
            type="submit"
          >
            Submit
          </button>,
        }
        <h4>
          All the tasks!
          <span>
             (
            2
            ) 
          </span>
        </h4>
        <ul>
          <li>
            <span>
              Just some demo tasks
            </span>
            <button
              type="button"
            >
              Edit
            </button>
            <button
              type="button"
            >
              delete
            </button>
          </li>
          <li>
            <span>
              As an example
            </span>
            <button
              type="button"
            >
              Edit
            </button>
            <button
              type="button"
            >
              delete
            </button>
          </li>
        </ul>
      </section>
    </main>
    <footer>
      <div
        class="css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="css-13cymwt-control"
        >
          <div
            class="css-hlgwow"
          >
            <div
              class="css-1dimb5e-singleValue"
            >
              Industrial
            </div>
            <div
              class="css-19bb58m"
              data-value=""
            >
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="color: inherit; background: ; opacity: 1; width: 100%; grid-area: 1 / 2; font: inherit; min-width: 2px; border: 0px; margin: 0px; padding: 0px; outline-color: initial; outline-style: initial; outline-width: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="css-1wy0on6"
          >
            <span
              class="css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
